File: A2Z_sheet/Array/Easy/03_check_if_array_is_sorted_and_rotated/solution.java
class Solution {
    public boolean check(int[] nums) {
        int count = 0;
        for(int i=1;i<nums.length;i++){
            if(nums[i] < nums[i-1]) count++;
        }
        if(count > 1) return false;
        return count == 0 ? true : nums[0] >= nums[nums.length-1];
    }
}

File: A2Z_sheet/Array/Easy/03_check_if_array_is_sorted_and_rotated/solution.rs
impl Solution {
    pub fn check(nums: Vec<i32>) -> bool {
        let mut count: u8 = 0;
        for i in 1..nums.len() {
            if(nums[i] < nums[i-1]) {
                count += 1;
            }
        }
        if count > 1 {
            return false
        }
        return count == 0 || nums[0] >= nums[nums.len() - 1];
    }
}
